### PR TITLE
[#1272] Make cypress tests independent from each other

### LINKED ISSUE #1272: Make Cypress tests independent from each other
We can use https://docs.cypress.io/api/plugins/before-run-api.html#Log-the-browser-and-the-number-of-specs-that-will-be-run to set up the env before running the tests. We need to:

1. create a user
2. connect a chatplugin channel
3. create a conversation

### PR DESCRIPTION
**Pull request by ogbeche77**
Fixes #1272

### CODE REVIEW

File: integration/ui/create_ui_conversation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  'Create UI Conversation',
+  {
+    baseUrl: 'http://airy.core/chatplugin/ui',
+  },
+  () => {
+    const channelId = '3502a0a7-933d-5410-b5fc-51f041146d89';
+    const messageChatplugin = 'Hello from Inbox!';
+
+    it('Creates a chat plugin conversation', () => {
+      cy.visit('/example?channel_id=' + channelId);
```
Comment (pascal-airy) [inline integration/ui/create_ui_conversation.spec.ts:13]: please also use Cypress.env('channelId')) here

File: integration/ui/connect_chatplugin.spec.ts [hunk 2]
```diff
@@ -20,16 +20,17 @@ describe('Connect chatplugin channel', () => {
...
-    cy.get(`[data-cy=${cyChannelsChatPluginFormBackButton}]`).click();
-    cy.get(`[data-cy=${cyChannelsChatPluginList}]`)
-      .children()
-      .filter(`:contains(${chatPluginName})`)
-      .should('have.length', 1);
+    cy.wait(500);
+    cy.url().should('include', '/channels/connected');
+    cy.get(`[data-cy=${cyChannelsFormBackButton}]`).click();
+    cy.wait(500);
+    cy.get(`[data-cy=${cyChannelsChatPluginList}]`).filter(':contains("Cypress Chatplugin")');
```
Comment (pascal-airy) [inline integration/ui/connect_chatplugin.spec.ts:34]:
```suggestion
    cy.get(`[data-cy=${cyChannelsChatPluginList}]`).filter(':contains(Cypress.env('chatPluginName'))');
```

File: integration/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "compilerOptions": {
+    "target": "es5",
+    "lib": ["es5", "dom"],
```
Comment (pascal-airy) [inline integration/tsconfig.json:4]: where is this file coming from?

File: integration/ui/connect_chatplugin.spec.ts (hunk 2 above)
Comment (pascal-airy) [reply integration/ui/connect_chatplugin.spec.ts:34]: this should always use the var

File: integration/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "compilerOptions": {
+    "types": ["cypress"]
+  },
+  "include": ["**/*.ts"]
+}
```
Comment (pascal-airy) [inline integration/tsconfig.json:6]: sorry @ogbeche77 I mean you can remove the whole file

Comment (ogbeche77) [reply integration/tsconfig.json:6]: >  "types": ["cypress"]

@pascal-airy  Oh, I see.I had intended to fix the `Cypress import error` currently shown by the editor. But I see we don't have that on develop

Comment (pascal-airy) [reply integration/tsconfig.json:6]: ah cool. But it didn't work?

File: integration/ui/create_tag.spec.ts [hunk 5]
```diff
@@ -5,25 +5,59 @@ import {
...
+    cy.visit('/');
+    cy.get(`[data-cy=${cyShowTagsDialog}]`).click();
+    cy.get(`[data-cy=${cyTagsDialogInput}]`).type(Cypress.env('tagName'));
+    cy.get(`[data-cy=${cyTagsDialogColorSelectorRed}]`).check({force: true});
+    cy.get(`[data-cy=${cyTagsDialogButton}]`).click();
+
+    cy.visit('/tags');
+    cy.get(`[data-cy=${cyTagsSearchField}]`).type(Cypress.env('tagName'));
+    cy.get(`[data-cy=${cyTagsTable}]`).children().children().its('length').should('be.gte', 2);
+    cy.get(`[data-cy=${cyTagsTable}]`).children().children();
```
Comment (pascal-airy) [inline integration/ui/create_tag.spec.ts:58]: looks like leftover from testing

File: integration/ui/filter_conversation.spec.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    cy.get(`[data-cy=${cyBubble}]`).click();
+    cy.get(`[data-cy=${cyInputbarTextarea}]`).type(Cypress.env('messageChatplugin'));
+    cy.get(`[data-cy=${cyInputbarButton}]`).click();
+
+    cy.visit('/');
+    cy.url().should('include', '/conversations');
+    cy.get(`[data-cy=${cyConversationList}]`).children().children().its('length').should('eq', 1);
+    cy.get(`[data-cy=${cySearchButton}]`).click();
+    cy.get(`[data-cy=${cySearchField}]`).get('input').type(Cypress.env('searchQuery'));
+  });
```
Comment (pascal-airy) [inline integration/ui/filter_conversation.spec.ts:48]: didn't we have an assertion here before?

File: integration/cypress.json [hunk 7]
```diff
@@ -1,10 +1,16 @@
 {
-    "baseUrl": "http://airy.core/ui",
+   "baseUrl": "http://localhost:8080/ui",
```
Comment (pascal-airy) [inline integration/cypress.json:2]: please revert this change

File: integration/ui/filter_conversation.spec.ts [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    cy.get(`[data-cy=${cyChannelsChatPluginList}]`).filter(`:contains("${Cypress.env('chatPluginName')}")`);
+
+    cy.visit('http://airy.core/chatplugin/ui/example?channel_id=' + Cypress.env('channelId'));
+    cy.get(`[data-cy=${cyBubble}]`).click();
+    cy.get(`[data-cy=${cyInputbarTextarea}]`).type(Cypress.env('messageChatplugin'));
+    cy.get(`[data-cy=${cyInputbarButton}]`).click();
+
+    cy.visit('/');
+    cy.url().should('include', '/conversations');
+    cy.get(`[data-cy=${cyConversationList}]`).children().children().its('length').should('eq', 1);
```
Comment (pascal-airy) [inline integration/ui/filter_conversation.spec.ts:45]: this should check for `great or equal to 1`

**Review by pascal-airy — APPROVED** [review]
giphy
